py/picca/io.py

```diff
@@ -1435,21 +1460,36 @@ def read_deltas(in_dir,
                                                                 '/*.fits.gz'))
     files = sorted(files)
 
+    arguments = [(f, from_image) for f in files]
+    pool = Pool(processes=nproc)
+    results = pool.starmap(read_delta_file, arguments)
+    pool.close()
+
     deltas = []
     num_data = 0
-    for index, filename in enumerate(files):
-        userprint("\rread {} of {} {}".format(index, len(files), num_data))
-        if from_image is None:
-            hdul = fitsio.FITS(filename)
-            deltas += [Delta.from_fitsio(hdu) for hdu in hdul[1:]]
-            hdul.close()
-        else:
-            deltas += Delta.from_image(filename)
-
-        num_data = len(deltas)
-        if max_num_spec is not None:
-            if num_data > max_num_spec:
+    for delta in results:
+        if delta is not None:
+            deltas += delta
+            num_data = len(deltas)
+            if (max_num_spec is not None) and (num_data > max_num_spec):
                 break
+    deltas = deltas[:max_num_spec]
+
+    # deltas = []
```

Comment:
Removed commented version or add a comment specifying that it is the old way of loading data